### DISCUSSION TITLE
How to Stop Chrome to pass cookies in every API calls while using with Cypress

### DISCUSSION THREAD

**Discussion by BIPLAVGHOSAL** [cypress-io/cypress#26577] [Questions and Help]
I am using Cypress 12.5.1 with ChromeV112 and it is passing all the existing cookies in every API call while visiting a webpage.

These additional Cookies in Request Headers are making the API calls fail.

I have added {chromeWebSecurity: false} as config also.

These cookies are not getting passed and working as expected in Firefox.

can anyone please suggest?

**Comment by portyu9** [comment]
Hi @BIPLAVGHOSAL 

`chromeWebSecurity: false` does not control whether Chrome sends cookies.

Cookies are attached to requests according to their `Domain`, `Path`, `Secure`, `SameSite`, etc. attributes. So the first thing I would check is **which cookies are being sent and where they came from** — especially the `Set-Cookie` responses that created them.

With normal Cypress test isolation, cookies are cleared between tests. If you specifically need to remove existing cookies inside a test, you can do it before visiting the page:

```js
cy.clearAllCookies()
cy.visit('/')
```

If the application/server creates the cookies again while loading, clearing them beforehand will not prevent later requests from receiving them. In that case, the better fix is usually to correct the cookie's `Domain`/`Path` configuration on the server.

For a test-only workaround, Cypress can also remove the outgoing `Cookie` header before selected API requests:

```js
cy.intercept(
  {
    url: 'https://api.example.com/**',
    middleware: true,
  },
  (req) => {
    delete req.headers.cookie
  }
)

cy.visit('/')
```

Register that intercept **before** the requests occur.

Be careful with this workaround: if the API legitimately uses a cookie for authentication or session state, deleting the entire `Cookie` header will remove that too. In that situation, remove only the unwanted cookie or fix its server-side scope instead.

Since you saw different behavior in Chrome and Firefox, the cookie names plus their `Domain`, `Path`, `SameSite`, and `Secure` attributes would be needed to determine exactly why the browsers behaved differently.

Please mark this answer as accepted if it helped, thank you.
